src/graphql/candyMachine.ts

```diff
@@ -203,29 +199,61 @@ const runUploadV2UsingHiddenSettings = async (
     hash: new Array(32).fill(0),
   };
 
-  const pubkeyFields = ['solTreasuryAccount', 'collection'];
+  // these fields come in as strings, but we need to transform them to pubkey objects
+  const pubkeyFields = [
+    'solTreasuryAccount',
+    'collection',
+    'splTokenAccount',
+    'splToken',
+  ];
+
   for (const field of pubkeyFields) {
     if (!!config[field]) {
+      console.log('updating ', field, 'key');
       config[field] = new PublicKey(config[field]);
     }
   }
 
   config['authority'] = walletKeyPair;
-  config['price'] = sol(config['price'] as number);
+  if (!!config['solTreasuryAccount']) {
+    config['price'] = sol(config['price'] as number);
+  } else {
+    const mintAcct = await connection.getAccountInfo(config['splToken']);
+    const mintInfo = MintLayout.decode(mintAcct.data) as MintInfo;
+    const price: SplTokenAmount = token(
+      config['price'] as number,
+      mintInfo.decimals,
+    );
+
+    // TODO(will): cleanup the API so we don't need to rename these arbitrarily
```

Comment:
a nice way to increase chances todos get done is to make a ticket and reference it in a PR comment or in the code

Reply:
I have found that leaving TODO's in code is useful when they are suggesting future improvements. They provide useful context for the next person who has to read the code, while not generating the tracking overhead of a ticket.